### DISCUSSION TITLE
Priorities for v2.0

### DISCUSSION THREAD

**Discussion by registerrier** [gammapy/gammapy#5619] [General]
The preparation of v2.0 is the core of the new round of development that is starting now.

We need to prioritize projects and tasks and assign them. I would like to start discussing the main projects regarding new features, code restructuring, new interfaces, infrastructure re-organization.  

Here are some on-going things:

### New features:
- Multidimensional priors and spectral unfolding. See [PR 4907](https://github.com/gammapy/gammapy/pull/4907) and [PR 5468](https://github.com/gammapy/gammapy/pull/5468/) as well as draft PR [5625](https://github.com/gammapy/gammapy/pull/5625)
- Introducing nested sampling and bayesian inference. See corresponding [PIG draft](https://github.com/gammapy/gammapy/pull/5499)
- Support for the future observation data model from CTAO or VODF. See a prototype [here](https://github.com/gammapy/ctao_data_model)
- Improving support for event types in `Makers`, `Datasets` and `Estimators`. See e.g. this draft PR [5511](https://github.com/gammapy/gammapy/pull/5511)
- Implementation of metadata propagation in the workflow. See e.g. this [project](https://github.com/orgs/gammapy/projects/13)
- Addition of sensitivity in `Estimator`. Support for joint sensitivities in `SensitivityEstimator`. See a first step in this PR [5361](https://github.com/gammapy/gammapy/pull/5361)
- Unbinned dataset and analysis. 
- IRFmodels and systematics.

### Improved interface
- Introducing an explicit fit statistic object to separate `Datasets` and `Models` see e.g. (https://github.com/gammapy/gammapy/pull/4237)
- Restructured analysis class, see[PR 3852](https://github.com/gammapy/gammapy/pull/3852)

### Improved infrastructure
- Re-organize CI. See e.g. issue [5587](https://github.com/gammapy/gammapy/issues/5587)
- Re-structure documentation.

This is likely incomplete. Anything you would like to add? 

We should come up with a credible plan with priorities. 
What do you think is a must-have, what is nice to have, what is for later?

**Comment by bkhelifi** [comment]
One should also discuss and rank the feature requests (issues) that are labelled `SAT req.`

**Comment by AtreyeeS** [comment]
In the context of systematics, we can also decide how to handle systematic errors in flux points, specially in the context of multi-instrument fitting https://github.com/gammapy/gammapy/issues/3517
